apps/web/src/modules/dao/components/AdminForm/AdminForm.schema.ts

```diff
@@ -28,6 +27,33 @@ export interface AdminFormValues {
 const twentyFourWeeks = 60 * 60 * 24 * 7 * 24
 const tenMinutes = 60 * 10
 
+const allocationSchema = Yup.object().shape({
+  founderAddress: addressValidationSchema,
+  allocationPercentage: Yup.number()
+    .transform((value) => (isNaN(value) ? undefined : value))
+    .required('*')
+    .integer('Must be whole number')
+    .max(100, '< 100')
+    .when('admin', (admin, schema) => {
+      if (!admin) return schema.min(1, '> 0') // (condition, errorMessage) - allocation represented as % must be greater than or equal to 0
+      return schema
+    }),
+  endDate: Yup.string()
+    .required('*')
+    .test(
+      'isDateInFuture',
+      'Must be in future',
+      (value: string | undefined, fieldData) => {
+        if (!value) return false
+        if (value === fieldData?.parent?.endDate) return true
```

Comment:
maybe add a comment here to explain that we don't need to check if the value is clean